datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1214,19 +1214,6 @@ pub struct CreateView {
     pub definition: Option<String>,
 }
 
-/// Types of files to parse as DataFrames
```

Comment:
Another potential way to write this (and maybe reduce the diff) would be to add a new enum variant like "Dynamic":

```rust
pub enum FileType {
    /// Newline-delimited JSON
    NdJson,
 ...
   /// Custom factory
   Dynamic(String),
}
```

This strategy might be more "idiomatic" as it encodes more information in the type system rather than a `String`

I don't feel strongly about this, I just wanted to offer it as an option

Reply:
I started with that solution, but I think it makes more sense to go:

1. enum (as it was)
2. string (with hard-coded values)
3. everything-as-a-TableProviderFactory (string with no hard-coded values)

If this PR is accepted, I will move immediately onto step #3.